src/go/rpk/pkg/config/config.go

```diff
@@ -143,6 +143,21 @@ func AvailableModes() []string {
 	}
 }
 
+// FileOrDefaults return the configuration as read from the file or
+// the default configuration if there is no file loaded.
+func (c *Config) FileOrDefaults() *Config {
+	if c.File() != nil {
+		cfg := c.File()
+		cfg.loadedPath = c.loadedPath
+		cfg.ConfigFile = c.ConfigFile
+		return cfg
+	} else {
+		cfg := Default()
+		cfg.ConfigFile = c.ConfigFile // preserve ConfigFile property.
```

Comment:
If c.File is nil, isn't the input config the default config? When would cfg.ConfigFile != c.ConfigFile?

Reply:
When passing a `--config` flag since we use this method after calling params.Load and the flag changes that, see:

https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/config/params.go#L237-L256